client/app/lib/components/buttons/DeleteButton.tsx

```diff
@@ -0,0 +1,46 @@
+import { useState } from 'react';
+import { IconButton, IconButtonProps } from '@mui/material';
+import Delete from '@mui/icons-material/Delete';
+import ConfirmationDialog from '../ConfirmationDialog';
+
+interface Props extends IconButtonProps {
+  disabled: boolean;
+  onClick: () => Promise<void>;
```

Comment:
Might be good to accommodate both:

```suggestion
  onClick: () => Promise<void> | void;
```

Reply:
onClick needs to return a Promise, otherwise we will get a typescript error when we chain finally to this function